include/wil/cppwinrt_register_com_server.h

```diff
@@ -0,0 +1,90 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+//! @file
+//! Utilities for making managing COM server easier
+
+#ifndef __WIL_CPPWINRT_REGISTER_COM_SERVER_DEFINED
+#define __WIL_CPPWINRT_REGISTER_COM_SERVER_DEFINED
+
+#if _MSVC_LANG >= 201703L
+
+#include <winrt/base.h>
+#include <vector>
+
+namespace wil::details
+{
+template <typename T>
+struct Factory : winrt::implements<Factory<T>, IClassFactory, winrt::no_module_lock>
+{
+    HRESULT __stdcall CreateInstance(IUnknown* outer, GUID const& iid, void** result) noexcept final
+    {
+        *result = nullptr;
+
+        if (outer)
+        {
+            return CLASS_E_NOAGGREGATION;
+        }
+
+        return winrt::make_self<T>().as(iid, result);
+    }
+
+    HRESULT __stdcall LockServer(BOOL) noexcept final
+    {
+        return S_OK;
+    }
+};
+
+template <typename T = void, typename... Rest>
+void register_com_server(std::vector<DWORD>& registrations)
+{
+    auto& back = registrations.emplace_back();
+    winrt::check_hresult(

```

Comment:
If this throws, you'll have a bogus value at the end of the vector. Note that the value will be `0` - I'm unsure if the system treats that as an invalid value that's okay to blindly revoke or not